programs/nodeos/main.cpp

```diff
@@ -119,7 +120,7 @@ int main(int argc, char** argv)
          .default_unix_socket_path = "",
          .default_http_port = 8888
       });
-      if(!app().initialize<chain_plugin, net_plugin, producer_plugin, resource_monitor_plugin>(argc, argv)) {
+      if(!app().initialize<blockvault_client_plugin, chain_plugin, net_plugin, producer_plugin, resource_monitor_plugin>(argc, argv)) {
```

Comment:
`blockvault_client_plugin` should be removed. It is an optional plugin.

Reply:
Actually, blockvault_client_plugin is mandatory. It uses wether blockvault_client_plugin::get() returns nullptr or not to check if the plugin is functional. The reason for the design is to make sure blockvault_client_plugin is initialized before chain plugin. There no control of plugin initialization order of optional plugin AFAIK.

Reply:
Really, I thought you could specify dependency even when optional. For example chain_api_plugin is dependent on chain_plugin and it is optional. I believe that is honored on init/startup.

Reply:
See application.hpp initialize and startup